onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidatePassword.kt

```diff
@@ -1,28 +1,36 @@
 package com.alexafit.onboardingauthdomain.useCase
 
-class ValidatePassword {
+import com.alexafit.onboardingauthdomain.model.domain.mapper.ValidatePasswordResult
 
-    operator fun invoke(password: String): Pair<String, Boolean> {
-        return when (password.isBlank()) {
-            true -> Pair(password.trim(), false)
-            false -> {
-                val trimmedPassword = password.trimEnd()
-                trimmedPassword.isPasswordValid()
-            }
+class ValidatePassword {
+    operator fun invoke(password: String): ValidatePasswordResult {
+        return if (password.isBlank()) {
+            ValidatePasswordResult(
+                passwordResult = password.trim(),
+                validPassword = false
+            )
```

Comment:
Also if I remember correctly the backend treats a space character as a normal special character, so passwords would be allowed to start/end with a space. In the rare case a user really chooses this, this code would accept the password but trim it which will cause the backend to save a different password than the user entered